.github/ISSUE_TEMPLATE/rfc.yml

```diff
@@ -0,0 +1,145 @@
+labels: ["rfc", "pending-decision"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for opening an RFC!
+        
+        RFCs are a highly-structured format for submitting change requests. All major changes to OpenBao should first go through an RFC, to have a place for the community to discuss.
+        
+        The template has a lot of fields and is inspired by the Rust RFC template. Please take your time to fill it out carefully.
+        
+        Working experimental PoCs in a draft Pull Request are a very welcome attachment to an RFC, especially ones that are easy to check out and take for a spin.
+
+  - type: textarea
+    id: summary
+    attributes:
+      label: Summary
+      description: |
+        One paragraph explanation of the feature.
+      placeholder:
+      value:
+    validations:
+      required: true
+
+  - type: textarea
+    id: problem-statement
+    attributes:
+      label: Problem Statement
+      description: |
+        Why are we doing this? What problem are we solving?
+      placeholder:
+      value:
+    validations:
+      required: true
+
+  - type: textarea
+    id: user-facing-description
+    attributes:
+      label: User-facing description
+      description: |
+        Please describe in detail how the feature would work from the perspective of somebody using:
+          - How can the feature be used?
+          - How would you explain the feature to somebody?
+          - Please provide examples of the feature in action.
+
+      placeholder:
+      value:
+    validations:
+      required: true
+
+  - type: textarea
+    id: technical-description
+    attributes:
+      label: Technical Description
+      description: |
+        Please describe in detail how the feature would work from a technical perspective:
+          - Which components of OpenBao would be involved in the change?
+          - How would the components need to change?
+          - Are there any interactions with other features that should be mentioned?
+          - Are the any edge cases of the feature that should be discussed?
+      placeholder:
+      value:
+    validations:
+      required: true
+
+  - type: textarea
+    id: rationale-and-alternatives
+    attributes:
+      label: Rationale and alternatives
+      description: |
+        - What would be the impact of this feature?
+        - Why is this solution better than alternative solutions to this problem, if there are any?
+        - How would the OpenBao user experience suffer if we didn't do this.
+      placeholder:
+      value:
+    validations:
+      required: true
+
+  - type: textarea
+    id: downsides
+    attributes:
+      label: Downsides
+      description: |
+        - Are there any disadvantages of implementing this?
+      placeholder:
+      value:
+    validations:
+      required: false
+
+  - type: textarea
+    id: security
+    attributes:
+      label: Security Implications
```

Comment:
I don't think its worth highlighting _everything_ in the RFC design, but perhaps adding user experience here would be a good catch-all bucket: e.g., developer experience integrating with this proposed change, operator experience trying to manage/configure it, ...

Just to make sure it gets thought about before dropping a (complex but useful) feature in...

Reply:
Fair enough, added ✅